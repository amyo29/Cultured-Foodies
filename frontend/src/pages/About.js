import React from 'react';
import {Button} from 'react-bootstrap';

function About() {
  return (
    <div >
      <header >
        about us
      </header>
    </div>
  );
}

export default About;