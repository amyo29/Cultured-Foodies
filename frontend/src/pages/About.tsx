import React from 'react';
// import {Button} from 'react-bootstrap';

function About() {
  return (
    <div>
      <h2>
        About
      </h2>
      <td>
        <p>
          <b>one</b> page
        </p>
        <ul>
          <li>description of the site, its purpose, its intended users</li>
          <li>explanation of the interesting result of integrating disparate data</li>
          <li>the group members</li>
          <li>for each member
          <ul>
              <li>name</li>
              <li>photo</li>
              <li>bio</li>
              <li>major responsibilities</li>
              <li>no. of commits</li>
              <li>no. of issues</li>
              <li>no. of unit tests</li>
            </ul>
          </li>
          <li>stats
          <ul>
              <li>total no. of commits</li>
              <li>total no. of issues</li>
              <li>total no. of unit tests</li>
            </ul>
          </li>
          <li>data
          <ul>
              <li>links to the data <b>sources</b></li>
              <li>description of <b>how</b> each was <b>scraped</b></li>
            </ul>
          </li>
          <li>tools
          <ul>
              <li>tools used</li>
              <li>describe their use</li>
              <li>special focus on optional tools that were <b>not</b> required</li>
            </ul>
          </li>
          <li>a link to the <a href="https://gitlab.com/cs373-group-11/cultured-foodies">GitLab</a> repo</li>
          <li>a link to the <a href="https://www.postman.com">Postman</a> API</li>
        </ul>
      </td>
    </div>
  );
}

export default About;
