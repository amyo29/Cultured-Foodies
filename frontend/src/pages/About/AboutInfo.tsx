import AWSLogo from "./Images/Tools/AWSLogo.png";
import GitLabLogo from "./Images/Tools/GitLabLogo.png";
import PostmanLogo from "./Images/Tools/PostmanLogo.png";
import ReactLogo from "./Images/Tools/ReactLogo.png";
import BootStrapLogo from "./Images/Tools/BootstrapLogo.png";
import NameCheapLogo from "./Images/Tools/Namecheap.png";

import RestCountriesLogo from "./Images/APIs/RestCountriesLogo.png";
import EdamamLogo from "./Images/APIs/EdamamLogo.png";
import NewscatcherLogo from "./Images/APIs/NewscatcherLogo.png";

import AmyImg from "./Images/Members/AmyImg.jpg";
import JoshuaImg from "./Images/Members/JoshuaImg.jpg";
import LucindaImg from "./Images/Members/LucindaImg.jpg";
import SiddheshImg from "./Images/Members/SiddheshImg.png";
import TimImg from "./Images/Members/TimImg.jpg";
import VishalImg from "./Images/Members/VishalImg.jpg";

const apiInfo = [
	{
		name: "REST Countries",
		img: RestCountriesLogo,
		description: "Used to retrieve country information",
		link: "https://restcountries.eu/",
	},
	{
		name: "Edamam",
		img: EdamamLogo,
		description: "Used to find dishes",
		link: "https://developer.edamam.com/",
	},
	{
		name: "Newscatcher API",
		img: NewscatcherLogo,
		description: "Used to retrieve food related news content",
		link: "https://newscatcherapi.com/",
	}
]

const toolsInfo = [
  {
    title: "AWS",
    img: AWSLogo,
    description: "Cloud hosting platform",
    link: "https://aws.amazon.com/",
  },
  {
    title: "GitLab",
    img: GitLabLogo,
    description: "GitLab Repository and CI/CD platform",
    link: "https://gitlab.com/",
  },
  {
    title: "Postman",
    img: PostmanLogo,
    description: "API and streamline collaboration platform",
    link: "https://www.postman.com/",
  },
  {
    title: "React",
    img: ReactLogo,
    description: "A JavaScript library for building user interfaces",
    link: "https://reactjs.org/",
  },
  {
    title: "BootStrap",
    img: BootStrapLogo,
    description:
      "A free and open-source CSS framework directed at responsive, mobile-first front-end web development",
    link: "https://getbootstrap.com/",
  },
  {
    title: "NameCheap",
    img: NameCheapLogo,
    description: "Domain name registrar",
    link: "https://www.namecheap.com/",
  },
];
const teamInfo = [
	{
		name: "Vishal Tak",
		username: "VishalTak14",
		email: "vishal.tak@shoplc.com",
		img: VishalImg,
		role: "Back-end",
		bio: "I'm a third year CS major at UT Austin. I'm from Austin, TX and in my free time I enjoy playing video games and watching movies.",
		linkedin: "https://www.linkedin.com/in/vishal-tak-898181163/",
		commits: 0,
		issues: 0,
		tests: 0,
	},
    {
		name: "Lucinda Nguyen",
		username: "lucinda",
		email: "lucinda.onguyen@gmail.com",
		img: LucindaImg,
		role: "Full-Stack",
		bio: "Bio description",
		linkedin: "",
		commits: 0,
		issues: 0,
		tests: 0,
	},
  {
		name: "Amy Ouyang",
		username: "amyo29",
		email: "yawen1999@gmail.com",
		img: AmyImg,
		role: "Full-Stack",
		bio: "I'm a 4th year CS Honors student at UT Austin. I love food.",
		linkedin: "https://www.linkedin.com/in/amy-o/",
		commits: 0,
		issues: 0,
		tests: 0,
	},
  {
		name: "Tim Nguyen",
		username: "Tim Nguyen",
		email: "nguyenmctim@gmail.com",
		img: TimImg,
		role: "Full-Stack",
		bio: "I'm a fourth year CS major at UT Austin. I'm from a small town called Palacios, TX and I enjoy playing video games and watching anime.",
		linkedin: "https://www.linkedin.com/in/nguyenmctim/",
		commits: 0,
		issues: 0,
		tests: 0,
	},
  {
		name: "Siddhesh Krishnan",
		username: "Siddhesh",
		email: "siddheshkrishnan1@gmail.com",
		img: SiddheshImg,
		role: "Full-Stack",
		bio: "Bio description",
		linkedin: "",
		commits: 0,
		issues: 0,
		tests: 0,
	},
	{
		name: "Joshua Andrew Arrojado",
		username: "jea2876",
		email: "joshuanms@gmail.com",
		img: JoshuaImg,
		role: "Full-Stack",
		bio: "I'm a fourth year CS student at UT Austin. I like to watch anime and kdramas in my sparetime. I also enjoy playing League!",
		linkedin: "https://www.linkedin.com/in/joshua-arrojado-803bb895/",
		commits: 0,
		issues: 0,
		tests: 0,
	},
];

export {apiInfo, toolsInfo, teamInfo};