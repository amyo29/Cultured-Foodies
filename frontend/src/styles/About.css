.font-style {
  font-family: serif;
  background-color: bisque;
}

.center {
  text-align: center;
}

.main-card {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
}

.profile-img {
  object-fit: cover !important;
	width: 350px !important;
	height: 350px !important;
	border-radius: 50% !important;
	margin: 25px !important;
	margin-bottom: 0px !important;
	display: block !important;
}