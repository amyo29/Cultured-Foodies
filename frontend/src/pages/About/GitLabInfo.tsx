import { teamInfo } from "./AboutInfo";

export const TOTAL_COMMITS_INDEX = 0
export const TOTAL_ISSUES_INDEX = 1
export const TOTAL_TESTS_INDEX = 2

const retrieveGitLabInfo = async () => {
  let statsInfo = [0, 0, 0]

  teamInfo.forEach((member: any) => {
    member.commits = 0;
    member.issues = 0;
    statsInfo[TOTAL_TESTS_INDEX] += member.tests;
  });

  const listOfIssues = await fetch(
    "https://gitlab.com/api/v4/projects/24676977/issues"
  ).then((response) => response.json());

  const listOfCommits = await fetch(
    "https://gitlab.com/api/v4/projects/24676977/repository/contributors"
  ).then((response) => response.json());

  listOfCommits.forEach((element: any) => {
    let name = element.name;
    let email = element.email;
    let commits = element.commits;
    teamInfo.forEach((member: any) => {
      if (
        member.name === name ||
        member.username === name ||
        member.email === email
      ) {
        member.commits += commits;
      }
    });
    statsInfo[TOTAL_COMMITS_INDEX] += commits;
  });

  listOfIssues.forEach((element: any) => {
    let name = element.author.name;
    teamInfo.forEach((member: any) => {
      if (member.name === name) {
        member.issues += 1;
      }
    });
    statsInfo[TOTAL_ISSUES_INDEX] += 1;
  });

  return {
    statsInfo: statsInfo,
    teamInfo: teamInfo,
  };
};

export default retrieveGitLabInfo;