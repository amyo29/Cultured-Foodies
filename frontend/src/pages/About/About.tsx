import React, { useEffect, useState } from "react";
import { API_INFO, TOOLS_INFO, TEAM_INFO } from "./AboutInfo";
import { InfoCard, ProfileCard } from "../../components/Card";
import retrieveGitLabInfo, {
  TOTAL_COMMITS_INDEX,
  TOTAL_ISSUES_INDEX,
  TOTAL_TESTS_INDEX,
} from "./GitLabInfo";
import "../../styles/About.css";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import Grid, { GridSpacing } from "@material-ui/core/Grid";
import { CardDeck } from "react-bootstrap";
import { Container, Row, Col, Card, ListGroup, Navbar } from "react-bootstrap";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    paper: {
      height: 140,
      width: 100,
    },
    control: {
      padding: theme.spacing(2),
    },
  })
);


function About() {
  useEffect(() => {
    document.title = "About"
  }, [])
  const [spacing, setSpacing] = React.useState<GridSpacing>(2);
  const classes = useStyles();

  const [commitsSum, changeCommitsSum] = useState(-1);
  const [issuesSum, changeIssuesSum] = useState(-1);
  const [testsSum, changeTestsSum] = useState(-1);
  const [teamData, changeTeamData] = useState([{}]);
  const [loaded, changeLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      if (teamData === undefined || teamData.length === 1) {
        const gitLabInfo = await retrieveGitLabInfo();
        changeCommitsSum(gitLabInfo.statsInfo[TOTAL_COMMITS_INDEX]);
        changeIssuesSum(gitLabInfo.statsInfo[TOTAL_ISSUES_INDEX]);
        changeTestsSum(gitLabInfo.statsInfo[TOTAL_TESTS_INDEX]);
        changeTeamData(gitLabInfo.TEAM_INFO);
        changeLoading(true);
      }
    };
    fetchData();
  }, [teamData]);

  const teamData1 = teamData.slice(0, 3);
  const teamData2 = teamData.slice(3, 6);
  var i, j;
  var chunk = 3;
  var toolRows = [];
  for (i = 0, j = TOOLS_INFO.length; i < j; i += chunk) {
    toolRows.push(TOOLS_INFO.slice(i, i + chunk));
  }
  return (
    <div className="font-style center">
      <h1>About Us</h1>

      <p>
        Cultured Foodies is a website that encourages its users to discover new
        foods from different cultures. We want people to explore the world of
        food from our website and learn more about new cuisines and dishes.
      </p>

      <p>
        The data we've gathered highlights cuisines from various cultures and
        their known presence in news outlets.
      </p>

      <h2>Meet the Team</h2>
      <div className="row main-card">
        <div className="col-7 card-deck">
            {loaded ? (
              teamData1.map((teamMember: any) => {
                const {
                  name,
                  img,
                  role,
                  bio,
                  commits,
                  issues,
                  tests,
                } = teamMember;

                return (
                  <ProfileCard
                    name={name}
                    img={img}
                    role={role}
                    bio={bio}
                    commits={commits}
                    issues={issues}
                    tests={tests}
                  />
                );
              })
            ) : (
              <div>Loading</div>
            )}
        </div>
      </div>
      <div className="row main-card">
        <div className="col-7 card-deck">
            {loaded ? (
              teamData2.map((teamMember: any) => {
                const {
                  name,
                  img,
                  role,
                  bio,
                  commits,
                  issues,
                  tests,
                } = teamMember;

                return (
                  <ProfileCard
                    name={name}
                    img={img}
                    role={role}
                    bio={bio}
                    commits={commits}
                    issues={issues}
                    tests={tests}
                  />
                );
              })
            ) : (
              <div>Loading</div>
            )}
        </div>
      </div>

      <div>
        <h2>Stats</h2>
        <div>
          {commitsSum === -1 ? (
            <h5>Total Commits: </h5>
          ) : (
            <h5>Total Commits: {commitsSum}</h5>
          )}
        </div>
        <div>
          {commitsSum === -1 ? (
            <h5>Total Issues: </h5>
          ) : (
            <h5>Total Issues: {issuesSum}</h5>
          )}
        </div>
        <div>
          {commitsSum === -1 ? (
            <h5>Total Tests: </h5>
          ) : (
            <h5>Total Tests: {testsSum}</h5>
          )}
        </div>
      </div>

      <div>
        <h2>API's</h2>
      </div>

      <Grid container className={classes.root} spacing={2} justify="center">
        {API_INFO.map((api: any) => {
          const { title, img, description, link } = api;

          return (
            <InfoCard
              title={title}
              img={img}
              description={description}
              link={link}
            />
          );
        })}
      </Grid>

      <div>
        <h2>Tools</h2>
      </div>


       <Container>
        {toolRows.map((cols) => (
          <Row>
            {cols.map((col: any, i: any) => (
              <Col>
                <InfoCard
                  title={col.title}
                  img={col.img}
                  description={col.description}
                  link={col.link}
                />
              </Col>
            ))}
          </Row>
        ))}
      </Container>
      <div>
        <h2>Our Data</h2>
      </div>
      <Grid container className={classes.root} spacing={2} justify="center">
        <InfoCard
          title="Gitlab Repo"
          img={TOOLS_INFO[1].img}
          link="https://gitlab.com/cs373-group-11/cultured-foodies"
        />
        <InfoCard
          title="Postman API"
          img={TOOLS_INFO[2].img}
          link="https://documenter.getpostman.com/view/14740527/Tz5i9gAY"
        />
      </Grid>
    </div>
  );
}

export default About;
