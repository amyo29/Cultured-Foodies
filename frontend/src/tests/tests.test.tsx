
import React from 'react';
//import { expect } from 'chai';
import * as Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import About from "../pages/About/About";

Enzyme.configure({
    adapter: new Adapter(),
})

test('renders the component', () => {
	const component = shallow(<About />);
	expect(component).toMatchSnapshot();
});

/* about page test 
it('Get About Page', async () => {
	const copy = shallow(<About />);
	expect(copy).to.not.be.undefined;
	expect(copy).to.have.length(1);
	expect(copy.find("div")).to.have.length(11);
});
*/
