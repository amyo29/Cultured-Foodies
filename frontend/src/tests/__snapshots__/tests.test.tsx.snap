// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render views About 1`] = `
<div
  className="font-style center"
>
  <h1>
    About Us
  </h1>
  <p>
    Cultured Foodies is a website that encourages its users to discover new foods from different cultures. We want people to explore the world of food from our website and learn more about new cuisines and dishes.
  </p>
  <p>
    The data we've gathered highlights cuisines from various cultures and their known presence in news outlets.
  </p>
  <h2>
    Meet the Team
  </h2>
  <Container
    fluid={false}
  >
    <div
      className="row"
    >
      <div>
        Loading
      </div>
    </div>
  </Container>
  <Container
    fluid={false}
  >
    <div
      className="row"
    >
      <div>
        Loading
      </div>
    </div>
  </Container>
  <div>
    <h2>
      Stats
    </h2>
    <div>
      <h5>
        Total Commits: 
      </h5>
    </div>
    <div>
      <h5>
        Total Issues: 
      </h5>
    </div>
    <div>
      <h5>
        Total Tests: 
      </h5>
    </div>
  </div>
  <div>
    <h2>
      API's
    </h2>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="row"
    >
      <InfoCard
        description="Used to retrieve country information"
        img="RestCountriesLogo.png"
        link="https://restcountries.eu/"
        title="REST Countries"
      />
      <InfoCard
        description="Used to find restaurant data."
        img="zomato.png"
        link="https://rapidapi.com/blog/how-to-use-zomato-api/"
        title="Zomato API"
      />
      <InfoCard
        description="Used to retrieve cuisine and dishes data."
        img="WorldFood.png"
        link="https://worldfood.guide/api/"
        title="WorldFood API"
      />
    </div>
    <div
      className="row"
    >
      <InfoCard
        description="Used to show the map location of a country"
        img="GoogleMapsLogo.jpeg"
        link="https://developers.google.com/maps/documentation/embed/get-started"
        title="The Maps Embed API"
      />
      <InfoCard
        description="Used to retrieve images of restaurants."
        img="GooglePlaceLogo.jpg"
        link="https://developers.google.com/maps/documentation/places/web-service/photos"
        title="Google Places Photos API"
      />
      <InfoCard
        description="Used to retrieve city data."
        img="teleportAPI.png"
        link="https://developers.teleport.org/api/"
        title="Teleport API"
      />
    </div>
  </Container>
  <div>
    <h2>
      Tools
    </h2>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="row"
    >
      <InfoCard
        description="Cloud hosting platform"
        img="AWSLogo.png"
        link="https://aws.amazon.com/"
        title="AWS"
      />
      <InfoCard
        description="GitLab Repository and CI/CD platform"
        img="GitLabLogo.png"
        link="https://gitlab.com/"
        title="GitLab"
      />
      <InfoCard
        description="API and streamline collaboration platform"
        img="PostmanLogo.png"
        link="https://www.postman.com/"
        title="Postman"
      />
    </div>
    <div
      className="row"
    >
      <InfoCard
        description="A JavaScript library for building user interfaces"
        img="ReactLogo.png"
        link="https://reactjs.org/"
        title="React"
      />
      <InfoCard
        description="A free and open-source CSS framework directed at responsive, mobile-first front-end web development"
        img="BootstrapLogo.png"
        link="https://getbootstrap.com/"
        title="BootStrap"
      />
      <InfoCard
        description="Domain name registrar"
        img="Namecheap.png"
        link="https://www.namecheap.com/"
        title="NameCheap"
      />
    </div>
    <div
      className="row"
    >
      <InfoCard
        description="Open-source relational database management system."
        img="PostgreSQL.png"
        link="https://postgresql.org"
        title="PostgreSQL"
      />
    </div>
  </Container>
  <div>
    <h2>
      Our Data
    </h2>
  </div>
  <Container
    fluid={false}
  >
    <div
      className="row justify-content-center"
    >
      <InfoCard
        img="GitLabLogo.png"
        link="https://gitlab.com/cs373-group-11/cultured-foodies"
        title="Gitlab Repo"
      />
      <InfoCard
        img="PostmanLogo.png"
        link="https://documenter.getpostman.com/view/14712963/Tz5jczL8"
        title="Postman API"
      />
    </div>
  </Container>
</div>
`;

exports[`Render views Cities 1`] = `
<div>
  <h1
    className="text-align center"
  >
    Cities
  </h1>
  <Container
    fluid={false}
  />
  <WithStyles(ForwardRef(Pagination))
    count={0}
    onChange={[Function]}
    page={0}
  />
</div>
`;

exports[`Render views Cuisines 1`] = `
<div>
  <h1
    className="text-align center"
  >
    Cuisines
  </h1>
  <Container
    fluid={false}
  />
  <WithStyles(ForwardRef(Pagination))
    count={0}
    onChange={[Function]}
    page={1}
  />
</div>
`;

exports[`Render views Restaurants 1`] = `
<div>
  <h1
    className="text-align center"
  >
    Restaurants
  </h1>
  <ForwardRef
    className="table"
    responsive={true}
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Name
        </th>
        <th>
          City
        </th>
        <th>
          Rating
        </th>
        <th>
          Price Range
        </th>
        <th>
          Average Cost for Two
        </th>
      </tr>
    </thead>
    <tbody />
  </ForwardRef>
  <WithStyles(ForwardRef(Pagination))
    count={0}
    onChange={[Function]}
    page={0}
  />
</div>
`;

exports[`Render views Splash Page 1`] = `
<div>
  <body
    className="Home"
    data-background="homephoto.jpg"
    id="mybody"
  >
    <header
      className="Home-header"
    >
      <div
        className="Title-summary"
      >
        Welcome fellow Foodie!
        <div
          className="summary"
        >
          Whether you are looking for cultural dishes
          <br />
          around the world, country information, dishes 
          <br />
          or global news, Cultured Foodies 
          <br />
          can help you find what you're looking for!
          <br />
        </div>
      </div>
    </header>
  </body>
</div>
`;
